src/core.rs

```diff
@@ -211,14 +196,19 @@ impl Wormhole {
     ///
     /// If the [`RendezvousServer`] is not properly initialized, i.e. if the
     /// mailbox is not open.
-    pub async fn connect_custom(
-        mut server: RendezvousServer,
-        appid: AppID,
-        password: String,
-        app_versions: impl serde::Serialize + Send + Sync + 'static,
+    pub async fn connect(
+        mailbox_connection: MailboxConnection<impl serde::Serialize + Send + Sync + 'static>,
```

Comment:
```suggestion
        MailboxConnection {
            config,
            mut server,
            mailbox: _mailbox,
            code,
            welcome: _welcome,
        }: MailboxConnection<impl serde::Serialize + Send + Sync + 'static>,
```

I think nowadays one can destructure function arguments like this? Not sure if it is an improvement over doing it in a let bindings below. (Also, you can use `..` to ignore all remaining fields without having to list them)

Reply:
Yes, did not now that. This works great, especially the `..`.